glusterd2/middleware/auth.go

```diff
@@ -30,6 +32,14 @@ func getAuthSecret(issuer string) string {
 	return ""
 }
 
+func generateQsh(r *http.Request) string {
```

Comment:
Can you please move this into `pkg/utils` or another common location. This would allow us to use the same function in the `restclient.setAuthToken` as well.